### PR TITLE
Update all-docs documentation index to Fast DDS [8747]

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

File: source/index.rst [hunk 1]
```diff
@@ -6,25 +6,23 @@
...
-optimum protocol, low cpu use, etc working under very different
-scenarios, from high-performance systems with high speed reliable links,
-to small devices connected over unreliable radio links.
+eProsima is focused in networking middleware and tools for high performance distributed systems and our products are
+customer driven.
+Over the years we have excelled in several areas of expertise: low latency, high throughput, small footprint, low
+bandwidth consumption, optimum protocol, low CPU use, etc., working under various scenarios, from high-performance
+systems with high speed reliable links, to small devices connected over unreliable radio links.
 
 .. image:: http://www.eprosima.com/images/boxes/Fast_RTPS_box200w.png
```
Comment (EduPonz) [inline source/index.rst]: Update images here

File: source/index.rst [hunk 2]
```diff
@@ -6,25 +6,23 @@
...
 ====================================
 
-eProsima is focused in networking middleware and tools for high
-performance distributed systems and our products are customer driven.
-Over the years we have excelled in several areas of expertise: Low
-latency, high throughput, small footprint, low bandwidth consumption,
-optimum protocol, low cpu use, etc working under very different
-scenarios, from high-performance systems with high speed reliable links,
-to small devices connected over unreliable radio links.
+eProsima is focused in networking middleware and tools for high performance distributed systems and our products are
```
Comment (EduPonz) [inline source/index.rst]:
```suggestion
eProsima is focused in developing networking middleware and tools for high performance distributed systems and our products are
```

File: source/index.rst [hunk 3]
```diff
@@ -6,25 +6,23 @@
...
-    :target: `eProsima Fast RTPS`_
+    :target: `eProsima Fast DDS`_
 
 ----------------------
-`eProsima Fast RTPS`_
+`eProsima Fast DDS`_
 ----------------------
 
-**eProsima Fast RTPS** is a high performance publish subscribe
+*eProsima Fast DDS* is a high performance publish subscribe
```
Comment (EduPonz) [inline source/index.rst:25]: In general, Link to standard?

File: source/index.rst [hunk 4]
```diff
@@ -35,10 +33,10 @@ based on Publishers, Subscribers, and Data Topics.
     :target: `eProsima Shape Demo`_
 
 ---------------------------------
-`eProsima Shape Demo`_
+`eProsima Shapes Demo`_
 ---------------------------------
 
-Shapes Demo is an application that Publishes and Subscribes to shapes of
+*eProsima Shapes Demo* is an application that Publishes and Subscribes to shapes of
```
Comment (EduPonz) [inline source/index.rst]: In general, mention Fast DDS here

File: source/index.rst [hunk 5]
```diff
@@ -136,23 +134,20 @@ application code, and the serialization support is generated at runtime.
...
 
-**eProsima Discovery Server** is an application that simplifies the use and testing of the new `Fast RTPS`_
+*eProsima Discovery Server* is an application that simplifies the use and testing of the new `eProsima Fast DDS`_
 discovery capabilities.
-Now Fast RTPS allows a centralized, non-standard, client-server discovery mechanism that:
+*eProsima Fast DDS* allows a centralized, non-standard, client-server discovery mechanism that:
 
-- extends Fast RTPS use to non-unicast network transports like TCP.
+- extends *eProsima Fast DDS* use to non-unicast network transports like TCP, and
 - speeds up the discovery-stage in a large number of participants scenarios.
```
Comment (EduPonz) [inline source/index.rst]: This list is not indented

File: README.md [hunk 6]
```diff
@@ -1,65 +1,72 @@
 # eProsima project documentation index
```
Comment (EduPonz) [inline README.md]: I'd put a readme on how to use the repo. Also, add the license from fastdds-docs

**Review by EduPonz — CHANGES_REQUESTED** [review]

File: source/index.rst (hunk 5 above)
Comment (rsanchez15) [reply source/index.rst]: It looks like this because that's where the image ends

File: README.md [hunk 7]
```diff
@@ -1,72 +1,102 @@
-# eProsima project documentation index
+# eProsima projects documentation index
```
Comment (EduPonz) [inline README.md:1]: In general
```suggestion
# eProsima documentation index
```

File: README.md [hunk 8]
```diff
@@ -1,72 +1,102 @@
...
 
-## <a href="https://eprosima-shapes-demo.readthedocs.io/en/latest/"><img alt="eProsima Shapes Demo Analytics" src="source/01-figures/Shapes_Demo.png" align="left" hspace="8" vspace="2" width="120" height="120"></a> [eProsima Shapes Demo Analytics](https://eprosima-shapes-demo.readthedocs.io/en/latest/)
+You can find all the library's source code on our [GitHub repository](https://github.com/eProsima/Fast-DDS).
 
-*eProsima Shapes Demo* is an application that Publishes and Subscribes to shapes of
-different colors and sizes moving on a board. Each Shape conforms its
-own topic: Square, Triangle or Circle. A single instance of the Shapes
-Demo can publish on or subscribe to several topics at a time.
+The documentation is built using [Sphinx](https://www.sphinx-doc.org), and it is hosted at [Read the Docs](https://readthedocs.org).
+The online documentation generated with this project can be found in [eProsima projects documentation index](https://docs.eprosima.com/).
```
Comment (EduPonz) [inline README.md:30]: Add links to sections of the readme

File: README.md [hunk 9]
```diff
@@ -1,72 +1,102 @@
...
-[eProsima Fast DDS](https://fast-dds.docs.eprosima.com/en/latest/) discovery capabilities.
-*eProsima Fast DDS* allows a centralized, non-standard, client-server discovery mechanism that:
-
-- extends *eProsima Fast DDS* use to non-unicast network transports like TCP, and
-- speeds up the discovery-stage in a large number of participants scenarios.
+```bash
+cd ~/all-docs
+source all-docs-venv/bin/activate
+make test
+```
```
Comment (EduPonz) [inline README.md:108]: I'm missing a contributing section linking to the file

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
+# Contribution Guidelines
```
Comment (EduPonz) [inline CONTRIBUTING.md:1]: This is AMAZING! :rocket:

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+
+*Fast DDS-Docs* is an open source project for documenting the *Fast DDS* library and some related tools such as *Fast DDS-Gen*.
+In that sense, contributions to expand or improve the documentation are most welcome and encouraged; we just want the best documentation possible for our community.
+
+There are two ways to contribute to this project:
+
+* [Opening an issue](#opening-an-issue).
+* [Submitting a pull request](#submitting-a-pull-request).
+
+Furthermore, this document also provide the contributor with a [Project structure](#project-structure) definition, and with a [Style guide](#style-guide) to ease the writing process.
```
Comment (EduPonz) [inline CONTRIBUTING.md]: No project structure here

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+
+If writing an issue does not do it for you because you just know what has to be done and how, and you have the energy and time to directly contribute to the documentation, submitting a pull request is your way to go.
+Please mind that by doing so, you accept to abide to the pull request process followed on this repository, meaning:
+
+1. Document one thing at a time.
+Avoid pull requests where several features or functionalities are documented at the same time.
+If you have many contributions, please split them in different pull requests.
+1. Submit you pull requests in separate branches.
+As for branch names, we normally use `features/*` for documentation expansions or refactors, `hotfix/*` for small fixes, and `bugfix/*` for larger parts that do not reflect reality and need to be corrected.
+1. Stick to the [Project structure](#project-structure) and [Style guide](#style-guide).
```
Comment (EduPonz) [inline CONTRIBUTING.md]: Same

**Review by EduPonz — CHANGES_REQUESTED** [review]

**Review by EduPonz — APPROVED** [review]

**Pull request merged**